crates/zed/src/zed/app_menus.rs

```diff
@@ -128,6 +128,7 @@ pub fn app_menus() -> Vec<Menu> {
                 MenuItem::action("Terminal Panel", terminal_panel::ToggleFocus),
                 MenuItem::separator(),
                 MenuItem::action("Diagnostics", diagnostics::Deploy),
+                MenuItem::action("Toggle Show All Diagnostics", editor::actions::ToggleDiagnostics),
```

Comment:
I think this action needs a rename: it's not clear that we toggle all diagnostics to be shown inline.
What is "toggling a diagnostic" even mean?

`ToggleShowAllDiagnostics` is better than the original, but can we do even better?
The new name also has to match the settings name.